Platform/OpenCanopy/BitmapFont.c

```diff
@@ -586,26 +602,55 @@ BmfGetTextInfo (
   return TextInfo;
 }
 
+STATIC
+EFI_GRAPHICS_OUTPUT_BLT_PIXEL
+mBlack = { 0, 0, 0, 255 };
+
+STATIC
+EFI_GRAPHICS_OUTPUT_BLT_PIXEL
+mWhite = { 255, 255, 255, 255 };
+
 STATIC
 VOID
-CopyInverted (
+BlendMem (
   EFI_GRAPHICS_OUTPUT_BLT_PIXEL *Dst,
-  EFI_GRAPHICS_OUTPUT_BLT_PIXEL *Src,
+  EFI_GRAPHICS_OUTPUT_BLT_PIXEL *AlphaSrc,
+  EFI_GRAPHICS_OUTPUT_BLT_PIXEL *Color,
   UINTN                         PixelCount
   )
 {
   UINTN  Index;
 
   for (Index = 0; Index < PixelCount; ++Index) {
-    Dst->Red       = 255 - Src->Red;
-    Dst->Green     = 255 - Src->Green;
-    Dst->Blue      = 255 - Src->Blue;
-    Dst->Reserved  = Src->Reserved;
+    // We assume that the font is generated by dpFontBaker
+    // and has only gray channel, which should be interpreted as alpha.
+    GuiBlendPixel(Dst, Color, AlphaSrc->Red);
     ++Dst;
-    ++Src;
+    ++AlphaSrc;
   }
 }
 
+STATIC CONST UINT8 clut[] = {
+  0x00, /* 0x00 0x00 0x00 white */
+  0xF6, /* 0x11 0x11 0x11 */
+  0xF7, /* 0x22 0x22 0x22 */
+  0x2A, /* 0x33 = 1*6^2 + 1*6 + 1 = 43 colors */
+  0xF8, /* 0x44 */
+  0xF9, /* 0x55 */
+  0x55, /* 0x66 = 2*(36 + 6 + 1) = 86 colors */
+  0xFA, /* 0x77 */
+  0xFB, /* 0x88 */
+  0x80, /* 0x99 = (3*43) = 129 colors*/
+  0xFC, /* 0xAA */
+  0xFD, /* 0xBB */
+  0xAB, /* 0xCC = 4*43 = 172 colors */
+  0xFE, /* 0xDD */
+  0xFF, /* 0xEE */
+  0xD6, /* 0xFF = 5*43 = 215 */
+};
+
+extern CONST UINT8 mAppleDiskLabelImagePalette[256];
```

Comment:
I think that needs to be in the header.

Reply:
And how it should be named, if it's no more static?

Reply:
Normally with `g` prefix (`gAppleDiskLabelImagePalette`).  Maybe `gAppleDiskLabelPalette` and `gAppleDiskLabelClut`?

Reply:
Clut now has no reason to be global

Reply:
GuiApp.h maybe?

Reply:
Looks more like OpenCanopy.h, because it's used in OpenCanopy.c and BitmapFont.c, and BitmapFont.c includes OpenCanopy.h and doesn't include GuiApp.h.

Reply:
Yeah, agree.